src/lib/votesReport.ts

```diff
@@ -100,8 +100,14 @@ class VotesReport extends Cache {
 
     if (typeof vote.choice !== 'number' && this.proposal) {
       choices = Array.from({ length: this.proposal.choices.length });
-      for (const [key, value] of Object.entries(vote.choice)) {
-        choices[parseInt(key) - 1] = value;
+      if (Array.isArray(vote.choice)) {
+        vote.choice.forEach((value, index) => {
+          choices[index] = value;
+        });
```

Comment:
```suggestion
        choices = vote.choice;
```

Reply:
For approval voting, there is a possibility that user can pick less number of choices 
```js
[3, 4]
```

Reply:
In this case, the other choices should be empty if we use `choices = vote.choice;` it will show wrong output on csv

Reply:
Better I will add tests for approval and quadratic 😄